addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -12803,4 +12803,21 @@ QUnit.module("Views", (hooks) => {
         await click(target.querySelector(".o_data_row .o_data_cell"));
         assert.containsOnce(target, ".o_form_view");
     });
+
+     QUnit.test('archive action not shown with readonly active field', async function (assert) {
+
+        // add active field on partner model, but do not put it in the view
+        this.data.partner.fields.active = {string: 'Active', type: 'char', default: true, readonly: true};
+
+         await makeView({
+            type: "form",
+            resModel: "partner",
+            resId: 1,
+            data: this.data,
+            arch: `<form><field name="foo"/></form>`,
+        });
+
+        assert.containsNone(target, '.o_cp_action_menus a:contains(Archive)');
```

Comment:
this assertion is false and does not have a safeguard if the action items become anything else than an `<a/>` tag.

You should instead:
- open the actions dropdown
- assert the list of action items